rcsb/utils/struct/EcodClassificationProvider.py

```diff
@@ -200,6 +201,26 @@ def __fetchFromSource(self, urlTarget):
         #
         return nmL
 
+    def __fetchFromBackup(self, urlTarget):
+        """Fetch and unzip the classification names and domain assignments from the backup file."""
+        fU = FileUtil()
+        fn = fU.getFileName(urlTarget)
+        fp = os.path.join(self.__dirPath, fn)
+        if not fU.exists(fp):
+            fU.get(urlTarget, fp)
+        #
+        with gzip.open(fp, "rt", encoding="utf-8-sig") as ifh:
+            line = ifh.readline()
+            line = ifh.readline()
+            line = ifh.readline()
+            ff = line[:-1].split()
+            self.__version = ff[-1]
+        #
+        nmL = self.__mU.doImport(fp, fmt="list", uncomment=True)
```

Comment:
So does this mean `mU.doImport` will read in a `.txt` and `.txt.gz` file the same way no problem? I was expecting this to be the part of the code that needed to be changed. If the only thing that is affected by the `.gz` compression is the extraction of the version information, then I would suggest just pulling out the version blocks into a separate function, with inputs of `(fileUrl, compressed=True/False)`, and calling it at lines 167-169 above. Then you could re-use `__fetchFromSource()` for both the Source and Fallback.

Reply:
Yes, `doImport` has handling for gzip already built in. I did have a similar thought to your suggestion but was undecided, so I figured I'd get your input first. I'll make the change. Having two methods is more consistent with other providers, but in this case there is a lot of redundant code.

As a side note, it seems annoying to have to open the file an extra time to get the version. But I'm not sure there's a simple way to get around that.

Reply:
Yeah I agree it's not very elegant to open the same file twice to get two different sets of information. Also not a fan with how deliberate the version-parsing is (i.e., read line 1, read line 2, read line 3, then get the version). It's definitely not ideal. but if there's no easy way around it (due to the `doImport` call removing all comments), we're at least not doing any more harm by keeping it that way...

Reply:
Your comment gave me an idea. Instead of opening the file twice, `doImport` will now not remove the comments. Then after getting the version, remove those items from the data list.

Reply:
Are there other lines in the file that have could have comments? It looks like that `uncomment=True` flag causes the removal of all in-line comments (even those occurring later in the line). So as long as that doesn't impact this case, then that sounds good to me!

Reply:
Yeah, actually I had the same thought and changed my original solution to this.:
`nmL = [line for line in nmL if not line.startswith("#")]`

Reply:
Hmm, well the difference between that and `uncomment=True` is that the latter will also remove comments that occur later in the line, such as:
```
data1 data2  # here is my comment
```

Reply:
My mistake, I misread later in the line as later in the file. I am looking at [`__processList` ](https://github.com/rcsb/py-rcsb_utils_io/blob/9f1f51aa2622e2ae4f799aaf7006226002be4e90/rcsb/utils/io/IoUtil.py#L537) and it's just doing a startswith. I think different file types are handled differently (for better or worse).

In any case, there are currently no # chars anywhere else in the file which is pretty much just a tab-delimited file.

Reply:
Ah OK, no my bad. I was just looking at [`uncommentFilter`](https://github.com/rcsb/py-rcsb_utils_io/blob/9f1f51aa2622e2ae4f799aaf7006226002be4e90/rcsb/utils/io/IoUtil.py#L76), and assumed that was what was being called. I didn't do a thorough dig.

So yeah, since there aren't any `#` further down in the file, then your method looks good to me!